src/web/mod.rs

```diff
@@ -236,6 +236,27 @@ impl MatchedRelease {
     }
 }
 
+fn semver_match<'a, F: Fn(&Release) -> bool>(
+    releases: &'a [Release],
+    req: &VersionReq,
+    filter: F,
+) -> Option<&'a Release> {
+    if let Some(release) = releases
+        .iter()
+        .filter(|release| filter(release))
+        .find(|release| req.matches(&release.version))
+    {
+        Some(release)
+    } else if req == &VersionReq::STAR {
+        // semver `*` does not match pre-releases.
+        // When someone wants the latest release and we have only pre-releases
+        // just return the latest prerelease.
+        return releases.iter().find(|release| filter(release));
+    } else {
+        None
+    }
```

Comment:
```suggestion
    if req == &VersionReq::STAR {
        // semver `*` does not match pre-releases.
        // When someone wants the latest release and we have only pre-releases
        // just return the latest prerelease.
        return releases.iter().find(|release| filter(release));
    } else if let Some(release) = releases
        .iter()
        .filter(|release| filter(release))
        .find(|release| req.matches(&release.version))
    {
        Some(release)
    } else {
        None
    }
```

I think in the first case, you'll never find the matching version since it's a "star request". So instead of iterating twice, I think it'd be better to switch the conditions order.

Reply:
your change would change the behaviour. 

In the first case (in my and the current order) we use `VersionReq.match` also for the `*` case. 

The `else` part is just a fallback if we don't find anything, then we just return the first matching element without checking the actual `VersionReq`

Reply:
Then please add this explanation as a code comment. :)

Reply:
I extended the existing comment a little